internal/user.go

```diff
@@ -62,3 +62,14 @@ func GenerateUserSettings(credentials *corev1.Secret, tags []topology.UserTag) (
 		HashingAlgorithm: rabbithole.HashingAlgorithmSHA512,
 	}, nil
 }
+
+func GenerateUserLimits(userLimits topology.UserLimits) rabbithole.UserLimitsValues {
+	userLimitsValues := rabbithole.UserLimitsValues{}
+	if userLimits.Connections > 0 {
+		userLimitsValues["max-connections"] = int(userLimits.Connections)
+	}
+	if userLimits.Channels > 0 {
+		userLimitsValues["max-channels"] = int(userLimits.Channels)
+	}
```

Comment:
Zero is a valid value in RabbitMQ. Negative values are valid too. If we want to enforce this lower bound for values, I would enforce it with validation in the CRD, instead of silently ignoring the values here.

Reply:
Yeah, I did something slightly different in the vhost limits, which I might move over to here. It's a bit more annoying, since it requires using nested pointers, but it solves the problem of distinguishing between unset and 0.